tfplan/json_plan.go

```diff
@@ -167,25 +171,36 @@ func attributes(value interface{}, address []string, res map[string]string, sche
 func readJSONResources(data []byte) ([]jsonResource, error) {
 	plan := jsonPlan{}
 	err := json.Unmarshal(data, &plan)
-
 	if err != nil {
 		return nil, err
 	}
-
 	var result []jsonResource
-
 	for _, resource := range plan.PlannedValues.RootModules.Resources {
 		resource.Module = "root"
 		result = append(result, resource)
 	}
-
 	for _, module := range plan.PlannedValues.RootModules.ChildModules {
-		name := strings.SplitAfterN(module.Address, ".", 2)[1]
-		for _, resource := range module.Resources {
-			resource.Module = name
-			result = append(result, resource)
-		}
+		result = append(result, resourcesFromModule(&module, 0)...)
 	}
-
 	return result, nil
 }
+
+func resourcesFromModule(module *childModule, level int) []jsonResource {
+	if level > maxChildModuleLevel {
```

Comment:
Is there a practical reason we need a max?

Reply:
This is to prevent potential denial of service from user input, which is becoming pretty common kinds of vulnerability.

Reply:
I relaxed the max to 10000 and add a log message if this condition is hit.